### ISSUE TITLE
Panel: shrink content mode (Pane)

### ISSUE THREAD

**Issue by Rafaelki** [OfficeDev/office-ui-fabric-react#7373]
**Describe the feature that you would like added**
If it is possible, I would like that the Panel component had a property to specify if the panel should open over the content (as it is now) or make the content shrink (when there is enough space). In the latest mode, the user could specify the selector of the content container. If it is not specified, the content container could be the children of the body.
I think this is especially useful when the page contains a list of entities in DocumentCards and the Panel contains the details of every entity. At the moment, if the Panel is not Blocking, the user can see the details quickly, without closing the Panel, but only for those entities that are not hidden behind the Panel. Shrinking the content would solve this little issue.

**What component or utility would this be added to**
Panel

**Have you discussed this feature with our team, and if so, who**
No

**Additional context/screenshots**
The functionality is similar to this: <https://material-ui.com/demos/drawers/#persistent-drawer>
or the "Page details panel" in SharePoint online.

**Comment by khmakoto** [comment]
@micahgodbolt wanted to loop you in to get the discussion on this feature request started.

**Comment by micahgodbolt** [comment]
We've certainly considered this option a number of times. The main challenge is that it requires not just rendering the Panel, but also changing the styles of the rest of the application. In the material UI example, this is accomplished by adding custom styles to the main section of the application that pushes it over the same width as the panel. So really, you can do this today by adding custom styles to your application when panels are open.

![image](https://user-images.githubusercontent.com/1434956/50019031-25d1b100-ff86-11e8-8cb5-03084e9f4ad4.png)

Long story short, as a standalone control, it's difficult to make this functionality in a way that scales, and works for a multitude of applications.

**Comment by Rafaelki** [comment]
Thank you for the explanation. It is OK for me to close this request.
